snakemake_wrapper_utils/snakemake.py

```diff
@@ -27,3 +27,29 @@ def is_arg(arg, cmd):
         return True
 
     return False
+
+
+def _get_unnamed_arguments(parameter_list):
+    """
+    Get unnamed arguments of snakemake.input or snakemake.output.
+    Find it as the first key in the parameter_list.
+
+    If no unnamed arguments are found, return an empty list.
+
+    Run as:
+        _get_unnamed_arguments(snakemake.input)
+
+    """
+    keys_with_positions = parameter_list._names
```

Comment:
I don't like that this uses private attibutes of NamedList. I think the functionality should rather be directly implemented in NamedList. Can you please file a PR in the main Snakemake repository?